### PR TITLE
fix: relax object and project name ref restrictions

### PR DESCRIPTION
**Pull request by gtarpenning**

### CODE REVIEW

File: weave-js/src/react.tsx [hunk 1]
```diff
@@ -504,7 +504,8 @@ export const isWeaveObjectRef = (ref: ObjectRef): ref is WeaveObjectRef => {
 // Entity name should be lowercase, digits, dash, underscore
 // Unfortunately many teams have been created that violate this.
 const PATTERN_ENTITY = '([^/]+)';
-const PATTERN_PROJECT = '([^\\#?%:]{1,128})'; // Project name
+const PATTERN_PROJECT = '([^:]{1,128})'; // any non / char is valid in project_name
```
Comment (tssweeney) [inline weave-js/src/react.tsx:507]: this should be `/` not `:`

Comment (gtarpenning) [reply weave-js/src/react.tsx:507]: 🤦🏻

**Comment by tssweeney** [comment]
Integrating this PR into https://github.com/wandb/weave/pull/2145

**Comment by gtarpenning** [comment]
> Integrating this PR into #2145

@tssweeney the first test is slightly broken 

`artifactName: 'a/b/c'` --> `artifactName: 'a __&/b/c`

**Pull request closed without merging**